torch/distributed/fsdp/flat_param.py

```diff
@@ -76,6 +82,31 @@ class FlatParamShardMetadata(NamedTuple):
     param_offsets: Tuple[Tuple[int, int], ...]
 
 
+# TODO (awgu): Prefix these with "Handle" for now to avoid circular imports and
+# inadvertent misuses; coalesce with those in fully_sharded_data_parallel.py
+# later
+class HandleShardingStrategy(Enum):
+    FULL_SHARD = auto()
```

Comment:
Can you add a comment for what each of these options mean?

Reply:
This `enum` matches the one called `ShardingStrategy` in `fully_sharded_data_parallel.py`. Before we land this PR, I plan to switch to directly using `ShardingStrategy` and de-duplicating this `HandleShardingStrategy`. I included it during development for convenience.